testing/README.md

```diff
@@ -0,0 +1,173 @@
+# Test Infrastructure
+
+This directory contains the Kubeflow test Infrastructure.
+
+This is a work in progress see [google/kubeflow#38](https://github.com/google/kubeflow/issues/38)
+
+The current thinking is this will work as follows
+
+  * Prow will be used to trigger E2E tests
+  * The E2E test will launch an Argo workflow that describes the tests to run
+  * Each step in the Argo workflow will be a binary invoked inside a container
+  * The Argo workflow will use an NFS volume to attach a shared POSIX compliant filesystem to each step in the 
+    workflow.
+  * Each step in the pipeline can write outputs and junit.xml files to a test directory in the volume
+  * A final step in the Argo pipeline will upload the outputs to GCS so they are available in gubernator
+
+## Accessing Argo UI
+
+You can access the Argo UI over the API Server proxy.
+
+We currently use the cluster
+
+```
+PROJECT=mlkube-testing
+ZONE=us-east1-d
+CLUSTER=kubeflow-testing
+NAMESPACE=kubeflow-test-infra
+```
+
+After starting `kubectl proxy` you can connect to it at
+
+Then you can connect to the UI via the proxy at
+
+```
+http://127.0.0.1:8001/api/v1/proxy/namespaces/kubeflow-test-infra/services/argo-ui:80/
+```
+
+TODO(jlewi): We can probably make the UI publicly available since I don't think it offers any ability to launch workflows.
```

Comment:
argo-ui has a feature and API to exec into running containers (enabled using `--enable-web-console` install option). This is disabled by default in alpha3 but be aware you would not want this enabled if the UI were to be made public facing.